container/shim/src/fetchers/lassie.js

```diff
@@ -167,11 +191,36 @@ async function getRequestedBlockFromCar(streamIn, streamOut, requestedCidV1, pat
         throw new Error(`Requested CID ${requestedCidV1} doesn't equal first block CID ${rootCid}.`);
       }
 
-      sendBlockResponse(streamOut, bytes, cidV1.toString())
+      sendBlockResponse(streamOut, bytes, cidV1.toString());
     } else {
       const cacheKey = cidToCacheKey(cidV1);
       blockCache.set(cacheKey, bytes);
     }
     count++;
   }
 }
+
+async function queueMetricsReport() {
```

Comment:
```suggestion
async function queueMetricsReport(newMetrics) {
  metrics.push(newMetrics)
```

If we add the metrics pushing here, it's in the future less likely we forget to call `queueMetricsReport` after modifying the metrics, or that we call it too often.